### PR TITLE
add a cloulens-suricata quickstart for azure

### PR DESCRIPTION
**Pull request by lucazacu-keys**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [X ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: cloudlens-suricata-ubuntu/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+        "addressPrefix": "10.0.0.0/16",
+        "subnetName": "Subnet",
+        "subnetPrefix": "10.0.0.0/24",
+        "storageAccountType": "Standard_LRS",
+        "publicIPAddressType": "Dynamic",
+        "virtualNetworkName": "CloudLensVNET",
+        "ubuntuOSVersion": "16.04.0-LTS",
+        "subnetRef": "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworkName'), variables('subnetName'))]",
+        "toolPublicIPAddressName": "suricatsPublicIP",
+        "toolStorageAccountName": "[concat(uniquestring(resourceGroup().id), 'tooldisk')]",
```
Comment (bmoore-msft) [inline cloudlens-suricata-ubuntu/azuredeploy.json]: flip the concat here so tooldisk comes first

File: cloudlens-suricata-ubuntu/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+        "toolDnsLabelPrefix": "[concat('suricata-',uniquestring(resourceGroup().id))]",
+        "toolNicName": "SuricataVMNic",
+        "tapPublicIPAddressName": "cloudlensPublicIP",
+        "tapStorageAccountName": "[concat(uniquestring(resourceGroup().id), 'tapdisk')]",
```
Comment (bmoore-msft) [inline cloudlens-suricata-ubuntu/azuredeploy.json]: flip the concat here so tapdisk comes first

File: cloudlens-suricata-ubuntu/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+                    "networkInterfaces": [
+                        {
+                            "id": "[resourceId('Microsoft.Network/networkInterfaces',variables('toolNicName'))]"
+                        }
+                    ]
+                },
+                "diagnosticsProfile": {
+                    "bootDiagnostics": {
+                        "enabled": true,
+                        "storageUri": "[concat(reference(concat('Microsoft.Storage/storageAccounts/', variables('toolStorageAccountName')), '2016-01-01').primaryEndpoints.blob)]"
```
Comment (bmoore-msft) [inline cloudlens-suricata-ubuntu/azuredeploy.json]: update apiVersion

File: cloudlens-suricata-ubuntu/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+                    "networkInterfaces": [
+                        {
+                            "id": "[resourceId('Microsoft.Network/networkInterfaces',variables('tapNicName'))]"
+                        }
+                    ]
+                },
+                "diagnosticsProfile": {
+                    "bootDiagnostics": {
+                        "enabled": true,
+                        "storageUri": "[concat(reference(concat('Microsoft.Storage/storageAccounts/', variables('tapStorageAccountName')), '2016-01-01').primaryEndpoints.blob)]"
```
Comment (bmoore-msft) [inline cloudlens-suricata-ubuntu/azuredeploy.json]: update apiVersion

File: cloudlens-suricata-ubuntu/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+                    }
+                }
+            }
+        },
+        {
+            "name": "[concat(parameters('toolVmName'), '/config')]",
+            "type": "Microsoft.Compute/virtualMachines/extensions",
+            "location": "[resourceGroup().location]",
+            "apiVersion": "2015-06-15",
```
Comment (bmoore-msft) [inline cloudlens-suricata-ubuntu/azuredeploy.json]: update apiVersion

File: cloudlens-suricata-ubuntu/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,382 @@
...
+                        "enabled": true,
+                        "storageUri": "[concat(reference(concat('Microsoft.Storage/storageAccounts/', variables('tapStorageAccountName')), '2016-01-01').primaryEndpoints.blob)]"
+                    }
+                }
+            }
+        },
+        {
+            "name": "[concat(parameters('toolVmName'), '/config')]",
+            "type": "Microsoft.Compute/virtualMachines/extensions",
+            "location": "[resourceGroup().location]",
```
Comment (bmoore-msft) [inline cloudlens-suricata-ubuntu/azuredeploy.json]: must be parameters('location')

File: cloudlens-suricata-ubuntu/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,382 @@
...
+                "publisher": "Microsoft.Azure.Extensions",
+                "type": "CustomScript",
+                "typeHandlerVersion": "2.0",
+                "autoUpgradeMinorVersion": true,
+                "settings": {
+                    "skipDos2Unix": false,
+                    "timestamp": 123456789
+                },
+                "protectedSettings": {
+                    "fileUris": ["[concat(parameters('_artifactsLocation'), 'scripts/setup-suricata.sh')]"],
```
Comment (bmoore-msft) [inline cloudlens-suricata-ubuntu/azuredeploy.json]: this needs the sasToken appended

File: cloudlens-suricata-ubuntu/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,382 @@
...
+                    "commandToExecute": "[concat('./setup-suricata.sh',  ' ', parameters('cloudlensProjectKey'))]"
+                }
+            }
+        },
+        {
+            "name": "[concat(parameters('tapVmName'), '/config')]",
+            "type": "Microsoft.Compute/virtualMachines/extensions",
+            "location": "[resourceGroup().location]",
```
Comment (bmoore-msft) [inline cloudlens-suricata-ubuntu/azuredeploy.json]: must be parameters('location')

File: cloudlens-suricata-ubuntu/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,382 @@
...
+            "apiVersion": "2015-06-15",
```
Comment (bmoore-msft) [inline cloudlens-suricata-ubuntu/azuredeploy.json]: update apiVersion

File: cloudlens-suricata-ubuntu/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,382 @@
...
+                "publisher": "Microsoft.Azure.Extensions",
+                "type": "CustomScript",
+                "typeHandlerVersion": "2.0",
+                "autoUpgradeMinorVersion": true,
+                "settings": {
+                    "skipDos2Unix": false,
+                    "timestamp": 123456789
+                },
+                "protectedSettings": {
+                    "fileUris": ["[concat(parameters('_artifactsLocation'), 'scripts/setup-tap.sh')]"],
```
Comment (bmoore-msft) [inline cloudlens-suricata-ubuntu/azuredeploy.json]: append sasToken

File: cloudlens-suricata-ubuntu/azuredeploy.parameters.json [hunk 11]
```diff
@@ -0,0 +1,15 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "cloudlensProjectKey" : {
+            "value": "GEN-UNIQUE[-32]"
```
Comment (bmoore-msft) [inline cloudlens-suricata-ubuntu/azuredeploy.parameters.json]: remove the [-32], its not valid - just use GEN_UNIQUE-32 if you want 32 chars

File: cloudlens-suricata-ubuntu/scripts/setup-tap.sh [hunk 12]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash -v
+sudo apt-get update -y
+sudo apt-get -y install tcpreplay
+sudo apt-get install -y docker.io
+sudo ip link add name CloudlensReplay type dummy
+sudo ifconfig CloudlensReplay up
+wget https://s3.amazonaws.com/cloudlens-automation/quickstart/malware.pcap
```
Comment (bmoore-msft) [inline cloudlens-suricata-ubuntu/scripts/setup-tap.sh]: move this to azure storage so it's on the same backbone or add retries on wget

**Comment by lucazacu-keys** [comment]
I have implemented the requested changes.

**Pull request merged**